File: assets/code-variants/ch13/25.java
@Test 
public void displayGreeting_renderUserName() {
    when(mockUserService.getUserName()).thenReturn("Fake User");
    userGreeter.displayGreeting(); // Call the system under test. 
    verify(userPrompter).setText(eq("Fake User"), any(), any());
}

@Test 
public void displayGreeting_timeIsMorning_useMorningSettings() {
    setTimeOfDay(TIME_MORNING);
    userGreeter.displayGreeting(); // Call the system under test. 
    verify(userPrompt).setText(any(), eq("Good morning!"), any());
    verify(userPrompt).setIcon(IMAGE_SUNSHINE);
}

File: assets/code-variants/ch13/25.py
def test_display_greeting_render_user_name():
    mock_user_service.get_user_name.return_value = "Fake User"
    user_greeter.display_greeting()  # Call the system under test.
    user_prompter.set_text.assert_called_with("Fake User", ANY, ANY)

def test_display_greeting_time_is_morning_use_morning_settings():
    set_time_of_day(TIME_MORNING)
    user_greeter.display_greeting()  # Call the system under test.
    user_prompt.set_text.assert_called_with(ANY, "Good morning!", ANY)
    user_prompt.set_icon.assert_called_with(IMAGE_SUNSHINE)
